cogs/gamerwords.py

```diff
@@ -52,23 +52,116 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
 # THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
+import random
 import collections
 import re
+import asyncio
+from string import ascii_letters
+
 import discord
 from discord.ext import commands
-import asyncio
 import unidecode
-import random
+
 
 #GAMER_REGEX = r'(?:[^s]|^)(n+\s*i+\s*g\s*g+\s*(?:e+\s*r+|(a)))(\s*s+)?'
 GAMER_REGEX = r'(b+\s*r+\s*u+\s*h+)'
 
 with open('data/catchphrases.txt') as f:
 	CATCHPHRASES = list(map(str.rstrip, f))
 
+
+class UnicodeAwareBruhReplacer:
+	def __init__(self):
+		# match related
+		self.start_index = -1
+		self.letter_check = [False] * 4
+		self.match_length = 0
+		self.spaces = 0
+
+	def reset(self):
+		self.start_index = -1
+		self.letter_check = [False] * 4
+		self.match_length = 0
+		self.spaces = 0
+
+	def replace(self, text):
+		self.reset()
+		matches = []
+		total_length = len(text)
+
+		for index, char in enumerate(text):
+			is_last_char = index + 1 == total_length
+			decoded = unidecode.unidecode(char)
+
+			# b is an end check
+			if decoded in ["r", "R", "u", "U", "h", "H"]:
```

Comment:
if you want a linear time check, you can just use `decoded in 'rRuUhH'`. But since you're ok with calling an expensive function (unidecode) in a tight loop anyway, you can probably just use `decoded.lower() in 'ruh'`.

Reply:
Or well, better yet, decoded in self.GAMER_WORD_ENDINGS, where GAMER_WORD_ENDINGS is a class variable defined as `frozenset('ruh')`